From 52d6cb4e741ce414497e3966d5b6c5263f402ab3 Mon Sep 17 00:00:00 2001
From: Mirza Krak <mirza.krak@northern.tech>
Date: Wed, 22 Aug 2018 14:15:24 +0200
Subject: [PATCH 08/10] add config_mender_defines.h

Adopted from file auto-generated by Yocto (meta-mender),
with custom configuration applied.

Signed-off-by: Mirza Krak <mirza.krak@northern.tech>
---
 include/config_mender_defines.h | 34 +++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 include/config_mender_defines.h

diff --git a/include/config_mender_defines.h b/include/config_mender_defines.h
new file mode 100644
index 0000000000..e2b75fa70b
--- /dev/null
+++ b/include/config_mender_defines.h
@@ -0,0 +1,34 @@
+/* This file is adopted from an auto-generated file from the meta-mender
+   layer.
+*/
+
+/* Shell variables */
+#define MENDER_BOOT_PART_NUMBER 1
+#define MENDER_BOOT_PART_NUMBER_HEX 1
+#define MENDER_ROOTFS_PART_A_NUMBER 2
+#define MENDER_ROOTFS_PART_A_NUMBER_HEX 2
+#define MENDER_ROOTFS_PART_B_NUMBER 3
+#define MENDER_ROOTFS_PART_B_NUMBER_HEX 3
+#define MENDER_UBOOT_STORAGE_INTERFACE "mmc"
+#define MENDER_UBOOT_STORAGE_DEVICE 0
+
+/* BB variables. */
+#define MENDER_STORAGE_DEVICE_BASE "/dev/mmcblk0p"
+#define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 0x100000
+#define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2 0x200000
+#define MENDER_ROOTFS_PART_A_NAME "/dev/mmcblk0p2"
+#define MENDER_ROOTFS_PART_B_NAME "/dev/mmcblk0p3"
+#define MENDER_MTD_UBI_DEVICE_NAME ""
+
+/* For sanity checks. */
+#define MENDER_BOOTENV_SIZE 0x4000
+#undef MENDER_BOOTENV_RANGE
+
+/* These are actually not used in the current setup but still populated
+   according to xaprw001 configuration
+*/
+#define MENDER_BOOT_KERNEL_TYPE "bootz"
+#define MENDER_KERNEL_NAME "zImage"
+#define MENDER_DTB_NAME "at91-xaprw001.dtb"
+#define MENDER_UBOOT_PRE_SETUP_COMMANDS ""
+#define MENDER_UBOOT_POST_SETUP_COMMANDS ""
-- 
2.18.0

